src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -4208,34 +4208,42 @@ void SmartScript::ProcessEvent(SmartScriptHolder& e, Unit* unit, uint32 var0, ui
             break;
         case SMART_EVENT_NEAR_PLAYERS:
         {
+            uint32 playerCount = 0;
             ObjectVector units;
             GetWorldObjectsInDist(units, static_cast<float>(e.event.nearPlayer.radius));
 
             if (!units.empty())
             {
-                if (!unit || unit->GetTypeId() != TYPEID_PLAYER)
-                    return;
+                for (WorldObject* unit : units)
+                {
+                    if (unit->GetTypeId() == TYPEID_PLAYER)
+                        playerCount++;
+                }
 
-                if (units.size() >= e.event.nearPlayer.minCount)
+                if (playerCount >= e.event.nearPlayer.minCount)
                     ProcessAction(e, unit);
```

Comment:
Would it be better to include this check inside the for loop?
In case of a large range, i.e. 100 worldObjects in a raid setting, I believe putting it inside the for loop would be less costly.
Although I don't know if that holds for every situation, and in case of no players, maybe it'd make things worse, actually, since the check occurs every time.

Reply:
Added it inside to also allow for INVOKER to be set